test/test_file.py

```diff
@@ -24,20 +28,60 @@ def write_bzImage(path: Path,
         f.write(version_line)
 
 
+def write_raw(path: Path,
+              version_line: bytes
+              ) -> None:
+    """Write a raw kernel binary at `path`, with `version_line`"""
+    with open(path, 'wb') as f:
+        f.write(0x210 * b'\0')
+        f.write(version_line)
+
+
+def write_compress(path: Path,
+                   version_line: bytes
+                   ) -> None:
+    """Write a gzip compressed raw kernel binary at `path`,
+    with `version_line`"""
+    with gzip.open(path, 'wb') as f:
+        # gzip would compress a string of 0s below 0x200,
+        # so we fill in some incompressible gibberish
+        s = b''
+        for i in range(1, 0xff):
+            m = hashlib.sha1()
+            m.update(i.to_bytes(1, 'little'))
+            s += m.digest()
+        f.write(s)
+        f.write(version_line)
+
+
 class KernelImageTests(unittest.TestCase):
     def setUp(self) -> None:
         self.td = tempfile.TemporaryDirectory()
 
     def tearDown(self) -> None:
         self.td.cleanup()
 
-    def test_read_internal_version(self) -> None:
+    def test_read_internal_version_bz(self) -> None:
         path = Path(self.td.name) / 'vmlinuz'
         write_bzImage(path, b'1.2.3 built on test')
         self.assertEqual(
             KernelImage(path).read_internal_version(),
             '1.2.3')
 
+    def test_read_internal_version_raw(self) -> None:
```

Comment:
Would be nice to move this into proper pytest-based case and parametrize it for different compressors.

Reply:
Sadly my knowledge of pytest only covers cursing at its output, no idea what you'd like to see here